src/main/scala/nl.knaw.dans.easy.solr4files/components/VaultIO.scala

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.solr4files.components
+
+import java.io.InputStream
+import java.net.{ URI, URL }
+
+import org.apache.commons.io.IOUtils.readLines
+import resource.ManagedResource
+
+import scala.collection.JavaConverters._
+import scala.util.Try
+import scala.xml.{ Elem, XML }
+
+trait VaultIO {
```

Comment:
in general you could do better on error handling in this class. `acquireAndGet` throws the errors. Use `.map(...).tried` instead. _Please note:_ if your `.map` already returns a `Try`, don't forget to do `flatten` after `tried`!